linera-views/tests/map_view_tests.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use linera_views::{
+    common::get_interval,
+    map_view::ByteMapView,
+    memory::create_memory_context,
+    views::{CryptoHashRootView, RootView, View},
+};
+use rand::{Rng, SeedableRng};
+use std::collections::BTreeMap;
+
+#[derive(CryptoHashRootView)]
+pub struct StateView<C> {
+    pub map: ByteMapView<C, u8>,
+}
+
+fn remove_by_prefix<V>(map: &mut BTreeMap<Vec<u8>, V>, key_prefix: Vec<u8>) {
+    let key_list = map
+        .range(get_interval(key_prefix.clone()))
+        .map(|x| x.0.to_vec())
+        .collect::<Vec<_>>();
+    for key in key_list {
+        map.remove(&key);
+    }
+}
+
+fn convert_map<V>(map: BTreeMap<Vec<u8>, V>) -> Vec<(Vec<u8>, V)> {
+    let mut vec = Vec::new();
+    for (key, value) in map {
+        vec.push((key, value));
+    }
+    vec
```

Comment:
```suggestion
    map.into_iter().collect()
```

But I'd just inline that.